crates/forge/tests/cli/test_cmd.rs

```diff
@@ -1048,3 +1048,83 @@ Traces:
 "#
     ]]);
 });
+
+const DETERMINISTIC_RANDOMNESS_TEST_CONTRACT: &str = r#"pragma solidity 0.8.24;
+import {Test, console2} from "forge-std/Test.sol";
+contract DeterministicRandomnessTest is Test {
+
+    function testDeterministicRandomUint() public {
+        // Values computed with seed=0xa1b2c3d4e5f6a7b8c9d0e1f2a3b4c5d6e7f8a9b0c1d2e3f4a5b6c7d8e9f0a1b2
+        assertEq(vm.randomUint(), 77081245346511865599562993006478104953925486436965894010918779131974790769683);
+        assertEq(vm.randomUint(), 14542802942652090507428621915178123789774624752096317033652316187323944929692);
+        assertEq(vm.randomUint(), 39147676640426175167070557719485824063633890234647166597297614898342291873676);
```

Comment:
Per [rand's docs](https://docs.rs/rand/latest/rand/rngs/struct.StdRng.html):
> The algorithm is deterministic but should not be considered reproducible due to dependence on configuration and possible replacement in future library versions.

I think we also don't want to make too many guarantees. So I would make this into a single test that `console.log`s the randomUints, run it twice (with `-vv`) and assert that the outputs are the same, and a third with a different seed that is different, so that we don't hardcode values.